### ISSUE TITLE
Immunizations V: Link recovery immunizations to recovered cases [2]

### ISSUE THREAD

**Issue by MateStrysewske** [hzi-braunschweig/SORMAS-Project#5969]
### Situation Description

Concept issue for immunizations: #4081

### Feature Description
   
* Add a button to the recovery section of immunizations with caption "Link case" that opens a popup window that allows users to search for an existing case; you can use a smiliar logic and window as for the "Search specific case" action in the case directory, but the user should only be able to search for cases that they have access to (so not all cases in the country), and only search for recovered cases of the same person with the same disease; adjust the info text accordingly
* When linking a case to an immunization, automatically update the following fields:
  * set the date of last infection to the symptom onset date of the case (report date if onset date is empty)
  * set the date of first positive test result to the result date of the first positive pathogen test associated to a sample of the chosen case with tested disease == disease of the case
  * set the date of recovery to the date of outcome
  * don't overwrite any of the dates with null if it doesn't exist for the case
* When the user selects a case in that window, set the case as the related case of the immunization and change the caption of the button to "Open case" with the short UUID in brackets 
* In the mobile app, this can look similar to linking resulting cases to contacts

**Comment by kwa20** [comment]
@MateStrysewske Some comments:
- If a case is linked in the recovery section, update the following date fields accordingly (don't overwrite existing entries with null):
    - date of last infection: symptom onset date if given, otherwise report date of the case
    - date of first positive test result: test result date of the earliest collected positive sample if given
    - date of recovery: date of outcome

> the user should only be able to search for cases that they have access to (so not all cases in the country), and only search for recovered cases; adjust the info text accordingly

- also restrict the case search to cases of the same person for the same disease

**Comment by bernardsilenou** [comment]
@MateStrysewske @kwa20 
When we initially discussed the extended immunization module, then idea was to replace the vaccination attributes of the case with the immunization. ie on the case edit form, there would be a link to create an immunization of  the person of the case.
If this is still valid, then we can activate this link for all entitle having vaccination attributes as long as the disease== disease of immunization.
- From case/contact/ ep, we can link to existing immunization or create new one
- From immunization, we can also link to existing case/ contact/ep

We do not even need to create a `direct-link` in db between other entities and immunization but rather between immunization and person and another one between case/contact/ep and person.

**Comment by MateStrysewske** [comment]
@bernardsilenou This is already covered in other issues related to this topic.

@kwa20 Updated the specifications with your suggestions.

**Comment by sergiupacurariu** [comment]
If we have the situation in which the case is found based on the conditions above but there is no PathogenTest usable to update the immunization fields do we still link the case to Immunization?

**Comment by sergiupacurariu** [comment]
this ticket can be merged only after #5905  is merged

**Comment by kwa20** [comment]
@MateStrysewske @sergiupacurariu There might have been a misunderstanding here in terms of the search field being used. I check the initial description and I'm sorry I didn't catch it earlier. As I've seen on the nightly build server, the search pop-up is the one used to search specific cases via the ID. The problem is that no normal user can benefit from this kind of search at this point on the form because they will most likely never have the ID of the exact possible previous infection at hand. Therefore, they need to see a display of the cases linked to the immunization person for them to select one instead.

**Comment by MateStrysewske** [comment]
@kwa20 Makes sense, but we'll have to do this in an upcoming sprint because it's basically a complete refactoring of this feature. Could you therefore please open a new issue for the change request?

**Comment by kwa20** [comment]
@MateStrysewske Ok, here the reference to the associated issue https://github.com/hzi-braunschweig/SORMAS-Project/issues/6690
